src/main/java/de/tum/in/www1/exerciseapp/domain/MultipleChoiceQuestion.java

```diff
@@ -56,6 +55,38 @@ public void setAnswerOptions(List<AnswerOption> answerOptions) {
     }
     // jhipster-needle-entity-add-getters-setters - Jhipster will add getters and setters here, do not remove
 
+    @Override
+    public ScoringStrategy getScoringStrategy() {
+        switch (getScoringType()) {
+            case ALL_OR_NOTHING:
+                return new ScoringStrategyAllOrNothing();
+            default:
+                throw new RuntimeException("Only Scoring Type ALL_OR_NOTHING is implemented yet!");
+        }
+    }
+
+    @Override
+    public boolean isAnswerPerfect(SubmittedAnswer submittedAnswer) {
+        if (submittedAnswer instanceof MultipleChoiceSubmittedAnswer) {
+            MultipleChoiceSubmittedAnswer mcAnswer = (MultipleChoiceSubmittedAnswer) submittedAnswer;
+            for (AnswerOption answerOption : getAnswerOptions()) {
+                boolean isSelected = false;
+                for (AnswerOption selectedOption : mcAnswer.getSelectedOptions()) {
+                    if (selectedOption.getId().longValue() == answerOption.getId().longValue()) {
+                        isSelected = true;
+                        break;
+                    }
+                }
+                if (answerOption.isIsCorrect() ^ isSelected) {
```

Comment:
Please don't use the ^ operator, it's not obvious. Also this method has a rather complex logic. Please document it properly. I also don't like the method name. What is perfect? I would rather say isAnwserCorrect. Shouldn't this be part of the ScoringStrategy?

Reply:
It's not part of the scoring strategy because wether or not an answer is 100% correct doesn't depend on how you score it. If and only if all answer options were selected correctly, the answer is 100% correct. I will rename it, remove the XOR operator and add some documentation.